fingerprintdb_cuda.cu

```diff
@@ -246,5 +247,47 @@ float FingerprintDB::tanimoto_similarity_cpu(const Fingerprint& fp1,
     return (float)common / (float)(total-common);
 }
 
+std::vector<int> fold_fingerprint(std::vector<int> &fp, const int factor)
+{
+    int n_arr = sizeof(int)*CHAR_BIT;
+    int arr[n_arr*fp.size()];
```

Comment:
We may have to rethink how to do this, it means we're going to need 32x as much memory to hold the bit values.  I suppose it's 32x the new folded size, but even if we're folding by 8 that's still 4x the original size.  If we switch from it being an `int` array to a `char` array we can still hold the value, but use 4x less memory that way as well.  I guess it might not be that bad, because we only need this extra memory for every fingerprint currently being processed...

Reply:
I see your point. Would using string of '0' and '1' be more efficient here?

Reply:
an array of `char` is 1 byte per bit, which is probably fine for this purpose (and also the same memory usage as a string of 0/1).  You get there just by changing it from `int arr` to `char arr`.